### ISSUE TITLE
enableLoggingForURLSessionConfiguration breaks NSURLCache

### ISSUE THREAD

**Issue by racer1988** [Instabug/Instabug-iOS#311]
### Steps to Reproduce the Problem

* Activate instabug Network Logging with  `[Instabug enableLoggingForURLSessionConfiguration:configuration];`
* Run the application with Instabug enabled.
* Do a network request with .useProtocolCachePolicy
* Go Offline and do the same network request (offline and cache only request with  `NSURLRequest.CachePolicy.returnCacheDataDontLoad` so the cache is always valid)
* NOTE: This works the first boot of the app, from the second boot, the issue occurs

### Expected Behavior

* NSURLSession is returning the cachedResponse

### Actual Behavior

* NSURLSession is unable to read the cache

### Instabug integration code

```
        Instabug.start(withToken: "REDACTED", invocationEvents: [.screenshot])
        
        NetworkLogger.enabled = true
        // Disable Crash Reporting
        CrashReporting.enabled = false
        IBGLog.printsToConsole = false
        
        // Disable Welcome Message
        Instabug.welcomeMessageMode = .disabled
        // Only show bug reporting (no chat, no feedback)
        BugReporting.promptOptions = [.bug]
        // Only need the comment
        BugReporting.invocationOptions = [.emailFieldHidden, .commentFieldRequired]
        
        // Enable User Steps
        Instabug.trackUserSteps = true
        
        // Repro Steps with screenshots
        // .enable is deprecated
        // Instabug.reproStepsMode = .enable
        
        // This might be disabled
        // Instabug.shouldCaptureViewHierarchy = true
        
        // 'autoScreenRecordingEnabled' is deprecated: AutoScreen recording is disabled please contact support for further details.
        // Instabug.autoScreenRecordingEnabled = true
        
        // Session Profiler is enabled by default
        //Instabug.sessionProfilerEnabled = true
        
        
        
        Instabug.willSendReportHandler = { report in
            let attributes = InstaBugSupport.userAttributes()
            for key in attributes.keys {
                report.setUserAttribute(attributes[key], withKey: key)
            }

            return report
        }

        // This is done for each session configuration we use
       [Instabug enableLoggingForURLSessionConfiguration:configuration1];
       [Instabug enableLoggingForURLSessionConfiguration:configuration2];
       [Instabug enableLoggingForURLSessionConfiguration:configuration3];
```

### SDK Version
Instabug `8.0.20`

### iOS Version
iOS 12.1

### Device Model
Any

### More Details

This seems similar to #209

**Comment by Kmohamed** [comment]
@racer1988 Yes we are aware of this bug and it's going to be released very soon.
you can use this custom build.
`pod 'Instabug', :git => 'https://github.com/Instabug/Instabug-iOS.git', :branch => 'fix-custom-session-configuration'`

**Comment by Kmohamed** [comment]
@racer1988 but we support only single custom session configuration. We will use the latest one only

**Comment by racer1988** [comment]
@Kmohamed Thanks for the info and the branch! 
Ok will work on the config.

it would be great to log multiple configurations tough ;)

**Comment by Kmohamed** [comment]
@racer1988 This will be very difficult we can not know from the `NSURLRequest` it's `NSURLSessionConfiguration`.
I will update you once the fix released.

**Comment by Kmohamed** [comment]
Unfortunately No, but will be included in the next release.

**Comment by racer1988** [comment]
@Kmohamed is this in 8.2.2? Thanks :D

**Comment by Kmohamed** [comment]
Unfortunately No 🙈
but definitely will be included in the next one

**Comment by Kmohamed** [comment]
@racer1988 We released a fix for your issue it's on [v8.2.3](https://github.com/Instabug/Instabug-iOS/releases/tag/8.2.3). 
And Sorry for being late

**Comment by racer1988** [comment]
Awesome Thanks!
